qualcommax: add Edgecore EAP101

alealexpro100 opened the pull request:
The Edgecore EAP101 is a wall/ceiling mountable AP. The AP can be powered by either PoE or AC adapter.

alealexpro100 commented:
Flashed [Z-Stack-firmware](https://github.com/Koenkk/Z-Stack-firmware/blob/master/coordinator/Z-Stack_3.x.0/bin/CC2652R_coordinator_20240710.zip), so now it works as coordinator with Z2M.
BUT It cannot be re-flashed with other firmware anymore. May be _ble_reset_ and _ble_backdoor_ pins were changed?

Firmware was converted from hex to binary using `objcopy --input-target=ihex --output-target=binary CC2652R_coordinator_20240710.hex CC2652R_coordinator_20240710.bin`.

robimarko commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6018-eap101.dts:
```diff
@@ -0,0 +1,471 @@
+		compatible = "gpio-leds";
+
+		led_wlan2g {
+			label = "green:wlan2g";
```
Use function, function-enumerator and color instead of label for LEDs

alealexpro100 replied:
Unfortunately, I could not find correct triggers `linux,default-trigger ` for both LED_FUNCTION_LAN and LED_FUNCTION_WAN (in OpenWRT sources). These LEDs still don't work working.
Also, I already added `function-enumerator` for both LED_FUNCTION_LAN, but it changed nothing.

robimarko commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6018-eap101.dts:
```diff
@@ -0,0 +1,471 @@
+	pinctrl-0 = <&hsuart_pins &btcoex_pins>;
+	pinctrl-names = "default";
+	dmas = <&blsp_dma 2>,
+		<&blsp_dma 3>;
```
Just keep these in one line.

robimarko commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6018-eap101.dts:
```diff
@@ -0,0 +1,471 @@
+	};
+
+	hsuart_pins: hsuart_pins {
+		mux {
```
No need for the mux subnode, just put everything under `hsuart_pins`.

Same goes for all others that just have a single subnode

robimarko commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6018-eap101.dts:
```diff
@@ -0,0 +1,471 @@
+		/*
+		 * U-boot looks for "n25q128a11" node,
+		 * if we don't have it, it will spit out the following warning:
+		 * "ipq: fdt fixup unable to find compatible node".
```
Do we even care since you are passing a fixed partition table anyway?

alealexpro100 commented:
> I guess your board is not TIP based, hence the need for U-Boot CLI?

Tried to google "TIP based" meaning, but could not find meaning of it.
Flashing using U-Boot is necessary because updating from stock firmware causes restoring of stock firmware (this device has 2 slots and after 3 reboots it switches slot).
So, if OpenWRT firmware is flashed using firmware stock (sysupgrade), it also creates `certificates` partition as second one, causing kernel to not find rootfs (we set ` root=/dev/ubiblock0_1`) on boot.

alealexpro100 committed:
qualcommax: add Edgecore EAP101

The Edgecore EAP101 is a wall/ceiling mountable AP. The AP can be
powered by either PoE or AC adapter.

Device info:
 - IPQ6018 SoC
 - 1GiB RAM
 - 128MiB NAND flash
 - 32MiB SPI NOR
 - 3 Ethernet ports (1x2.5G Passive PoE, 2x1G non-PoE)
 - 1 Console port
 - 2GHz/5GHz AX1800 WLAN
 - 1 USB 2.0 port (not USB 3.0!)
 - CC2652 ZigBee, BLE module

Install instructions:

Prerequistes:
  - TFTP server (for example, tftpd64) and DHCP server
  - Console cable plugged in (115200 8N1 no flow control).

1. Power on device and interrupt u-boot (by pressing Enter) to obtain u-boot CLI

2. Get ip from dhpc, set serverip to IP address of the TFTP server:

	`dhcp && setenv serverip 192.168.1.250`

3. Backup original firmware:

	`nand device 0
	 nand read 0x44000000 0x3c00000 0x3c00000
	 tftpput 0x44000000 0x3c00000 eap101-fs2.img
     nand read 0x44000000 0x0 0x3c00000
     tftpput 0x44000000 0x3c00000 eap101-fs1.img`

4. Download image from TFTP server:

	`tftpboot 0x44000000 openwrt-qualcommax-ipq60xx-edgecore_eap101-squashfs-factory.ubi`

5. Flash ubi image to both partitions and reset:

	`imxtract 0x44000000 ubi
	 nand device 0
	 nand erase 0x0 0x3c00000
	 nand write $fileaddr 0x0 $filesize
	 nand erase 0x3c00000 0x3c00000
	 nand write $fileaddr 0x3c00000 $filesize
	 reset`

NOTES:
* If you encounter ECC errors, do ubiformat of flashing rootfs partition.
* To flash cc2652 (BSL - DIO_9), use `tisbl.sh /dev/ttyMSM1 115200 2652 /etc/tifirmware/<fw>.bin 79 34`.

Signed-off-by: Aleksey Nasibulin <alealexpro100@ya.ru>

alealexpro100 commented:
Returned usage of aliases for `led_wanpoe`, `led_lan1` and `led_lan2` because `LED_FUNCTION_WAN` and `LED_FUNCTION_LAN` seem to be NOT working alone.

robimarko commented:
> Returned usage of aliases for `led_wanpoe`, `led_lan1` and `led_lan2` because `LED_FUNCTION_WAN` and `LED_FUNCTION_LAN` seem to be NOT working alone.

On they own they dont assign a trigger, you still gotta add one via the 01_leds script if there is not an DT trigger available
